interfaces/Portal/src/app/components/update-fsp/update-fsp.component.ts

```diff
@@ -73,22 +69,24 @@ export class UpdateFspComponent implements OnInit {
   public selectedFspName: string;
   public attributesToSave: object = {};
   public enableUpdateBtn = true;
+  public reason = 'Financial service provider change';
 
   ngOnInit() {
     this.propertyModel = this.value;
     this.startingFspName = this.value;
-    this.getFspAttributes(this.value);
+    this.setSelectedFspAndPrepareDropdown(this.value);
     this.startingAttributes = [...this.selectedFspAttributes];
     this.enableUpdateBtn = false;
   }
 
   public updateChosenFsp() {
     this.programsService
-      .updateChosenFsp(
-        this.referenceId,
+      .updatePaAttribute(
         this.programId,
+        this.referenceId,
+        'programFinancialServiceProviderConfigurationName', // Not sure how this should be done in a strongly typed way in our Portal way of working
```

Comment:
If there is no `enum` with this attribute-name, then this is as good as can be, I think...

Or you'd have to "create string version of the `programFinancialServiceProviderConfigurationName`-property of a specific (return-)type"... (You can ask Copilot that, probably. :D )